framework/src/mesh/MooseMesh.C

```diff
@@ -1299,38 +1311,41 @@ MooseMesh::detectPairedSidesets()
 
```

Comment:
This routine only gets called by the periodic bc code. No need to over optimize.